osquery/events/linux/auditdnetlink.cpp

```diff
@@ -668,6 +579,155 @@ bool AuditdNetlink::configureAuditService() noexcept {
+
+  for (size_t read_retry = 0U; read_retry < 3U; ++read_retry) {
+    if (timeout < getUnixTime()) {
+      VLOG(1) << "Failed to unconfigure the audit service (timeout)";
+      return false;
+    }
+
+    bool netlink_ready = false;
+
+    for (size_t poll_retry = 0U; poll_retry < 3U; ++poll_retry) {
+      pollfd fds[] = {{audit_netlink_handle_, POLLIN, 0}};
+
+      errno = 0;
+      int poll_status = ::poll(fds, 1, 4);
+      if (poll_status == 0) {
+        continue;
+      }
+
+      if (poll_status < 0) {
+        VLOG(1) << "poll() failed with errno " << errno;
+        return false;
+      }
+
+      if ((fds[0].revents & POLLIN) != 0) {
+        netlink_ready = true;
+        break;
+      }
+    }
+
+    if (!netlink_ready) {
+      VLOG(1) << "Could not read from the audit netlink";
+      return false;
+    }
+
+    // Get the reply from the audit link
+    struct audit_reply reply = {};
+    if (audit_get_reply(
+            audit_netlink_handle_, &reply, GET_REPLY_NONBLOCKING, 0) <= 0) {
+      continue;
+    }
+
+    read_retry = 0;
+    if (reply.nlh->nlmsg_seq != static_cast<unsigned int>(seq)) {
+      continue;
+    }
+
+    if (reply.type == NLMSG_DONE) {
+      // We have finished listing the rules
+      break;
+    }
+
+    if (reply.type == NLMSG_ERROR && reply.error->error != 0) {
+      return false;
+    }
+
+    if (reply.type != AUDIT_LIST_RULES) {
+      // Skip this reply if it is not part of the rule list output
+      continue;
+    }
+
+    // Save the rule
+    auto reply_size = sizeof(reply) + reply.ruledata->buflen;
+
+    AuditRuleDataObject reply_object;
+    reply_object.resize(reply_size);
+    if (reply_object.size() != reply_size) {
+      VLOG(1) << "Failed to read the audit rule data";
+      return false;
+    }
+
+    std::memcpy(reply_object.data(), reply.ruledata, reply_size);
+    rule_object_list.push_back(reply_object);
+  }
+
+  // Delete each rule
+  size_t error_count = 0U;
+  for (auto& rule_object : rule_object_list) {
+    if (!deleteAuditRule(rule_object)) {
+      error_count++;
+    }
+  }
+
+  if (error_count != 0U) {
+    VLOG(1) << error_count << " audit rules could not be correctly removed";
+    return false;
+  }
+
+  VLOG(1) << "The audit configuration has been cleared";
+  return true;
+}
+
+bool AuditdNetlink::deleteAuditRule(const AuditRuleDataObject& rule_object) {
+  if (NLMSG_SPACE(rule_object.size()) > MAX_AUDIT_MESSAGE_LENGTH) {
+    return false;
+  }
+
+  auto rule_data =
+      reinterpret_cast<const struct audit_rule_data*>(rule_object.data());
+
+  struct audit_message request = {};
+  request.nlh.nlmsg_len = static_cast<__u32>(NLMSG_SPACE(rule_object.size()));
+  request.nlh.nlmsg_type = AUDIT_DEL_RULE;
+  request.nlh.nlmsg_flags = NLM_F_REQUEST;
+  std::memcpy(NLMSG_DATA(&request.nlh), rule_data, rule_object.size());
+
+  struct sockaddr_nl address = {};
+  address.nl_family = AF_NETLINK;
+
+  bool success = false;
+
+  for (size_t i = 0; i < 3; i++) {
```

Comment:
nit, about the 3-strikes loop calls the iterator variable "_retry" and uses the `U` literal type.